chainreg/chainregistry.go

```diff
@@ -652,18 +652,29 @@ func NewChainControl(cfg *Config) (*ChainControl, error) {
 	lnWallet, err := lnwallet.NewLightningWallet(walletCfg)
 	if err != nil {
 		fmt.Printf("unable to create wallet: %v\n", err)
-		return nil, err
+		return nil, nil, err
 	}
 	if err := lnWallet.Startup(); err != nil {
 		fmt.Printf("unable to start wallet: %v\n", err)
-		return nil, err
+		return nil, nil, err
 	}
 
 	log.Info("LightningWallet opened")
 
 	cc.Wallet = lnWallet
 
-	return cc, nil
+	ccCleanup := func() {
```

Comment:
Looks like we'd also want to return a closure to shutdown any subsystems already started in the event of an error before getting here.

Reply:
Right.
I removed the call to `cc.ChainView.Stop()` as its lifecycle is managed by the router and the code now returs the ccCleanup on previous errors.